geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/DeltaWriter.scala

```diff
@@ -320,9 +320,9 @@ object DeltaWriter extends StrictLogging {
       private var mappings: Map[String, java.util.Map[Integer, Integer]] = _
       private var count = 0 // records read in current batch
 
-      override def hasNext: Boolean = synchronized { count < toLoad.reader.getValueCount || loadNextBatch() }
+      override def hasNext: Boolean = count < toLoad.reader.getValueCount || loadNextBatch()
```

Comment:
Overall, can you remind me why we think that the removal of the 'synchronized' calls is ok now?

Reply:
i believe that we only needed the synchronized b/c we were calling next/hasNext/close in different threads. now they will all be called by the same thread, so we shouldn't need to synchronize. the low-level scan will be closed by the thread management, but not this iterator.

Reply:
Cool.  So someone would have to go to trouble to pass around GeoMesaReaders to multiple threads in their application and start calling methods like hasNext, next, and close in concurrent manners...  

If someone does that in their application, we cannot help them.  Is that the story?

Reply:
right